lib/action_view/component/content_for.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentFor
+      def content_for(attr, content = nil, options = {}, &block)
+        unless respond_to?(attr)
+          raise StandardError.new "Called content_for with unknown attribute '#{attr}'"
+        end
+
+        if block_given?
+          options = content if content
+          content = view_context.capture(&block)
+        end
+
+        existing_content = send(attr)
```

Comment:
I wonder if we should support this case. I wouldn't expect components to set content multiple times.

Reply:
yeah I could go either way. It's fair to argue the case for setting content multiple times in the component view must be as a result of logic or loops in the view that might be better encapsulated in the Ruby component file. I wonder if the case of looping would be reasonable. 

Let's decide I'm not fussed. It's probably more docs to write. Easy enough to add later with a real use case behind it.

Reply:
killed it